docs/pages/eas-build-custom-builds/custom-build-functions.mdx

```diff
@@ -0,0 +1,288 @@
+    - run:
+        name: Install dependencies
+        command: npm install
+    # @info #
+    - my_function:
+        inputs:
+          num1: 1
+          num2: 2
+        id: sum_function
+    # @end #
+    - run:
+        name: Print the sum
+        inputs:
+          sum: ${ steps.sum_function.sum }
+        command: echo ${ inputs.sum }
+    - run:
+        name: Finished
+        command: echo Finished
+
+functions:
+  my_function:
+    name: My function
+    # @info #
+    inputs:
+      - name: num1
+        type: number
+      - name: num2
+        type: number
+    outputs:
+      - name: sum
+    # @end #
+    path: ./myFunction
+```
+
+```ts .eas/build/myFunction/src/index.ts
+// This file was autogenerated by `create-eas-build-function` command.
+// Go to README.md to learn more about how to write your own custom build functions.
+
+import {
+  BuildStepContext,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+} from '@expo/steps';
+
+/* @info */
+interface FunctionInputs {
+  // first template argument is the type of the input value, second template argument is a boolean indicating if the input is required
+  num1: BuildStepInput<BuildStepInputValueTypeName.NUMBER, true>;
+  num2: BuildStepInput<BuildStepInputValueTypeName.NUMBER, true>;
+}
+/* @end */
+
+/* @info */
+interface FunctionOutputs {
+  // template argument is a boolean indicating if the output is required
+  sum: BuildStepOutput<true>;
+}
+/* @end */
+
+async function myFunction(
+  ctx: BuildStepContext,
+  {
+    inputs,
+    outputs,
+  }: // env,
+  {
+    inputs: FunctionInputs;
+    outputs: FunctionOutputs;
+    // env: BuildStepEnv;
+  }
+): Promise<void> {
+  /* @info */
+  ctx.logger.info(`num1: ${inputs.num1.value}`);
+  ctx.logger.info(`num2: ${inputs.num2.value}`);
+
+  const sum = inputs.num1.value + inputs.num2.value;
+
+  ctx.logger.info(`sum: ${sum}`);
+
+  outputs.sum.set(sum.toString()); // for now outputs can only of string type, but that will change in the future
+  /* @end */
+}
+
+export default myFunction;
+```
+
+<ImageSpotlight
+  alt="Example of a custom build config using custom JavaScript/TypeScript function."
+  src="/static/images/eas-build/custom-build-function-advanced.png"
+/>
+
+> **Info** Remember to rebuild your custom function after making any changes to it.
+
+</Step>
+
+## Summary
+
+- Custom JavaScript/TypeScript functions are a great way to extend the EAS Build custom builds functionality with your custom logic, in a pleasant way, by writing JavaScript/TypeScript code.
+- Custom JavaScript/TypeScript functions are reusable and can be used in multiple workflows.
+- Using custom JavaScript/TypeScript functions is a great option for more advanced use cases that are not easy to do by writing shell scripts.
+- Most of our [built-in functions](/eas-build-custom-builds/custom-build-config-schema/#built-in-eas-functions) are open-sourced and can be used as a reference for writing your own custom functions.
```

Comment:
Need to change the URL here if we/when we change the filenames and address the first suggestion.